[Single Machine Performance] Nightly Workload Checks

blt opened the pull request:
### What does this PR do?

This commit introduces the functional test that drives Workload Checks in the Agent project. Tests for a 'typical' machine and its workloads are also added.

### Motivation 

Single Machine Performance intend to do historical analysis of the Agent's fitness. The Workload Checks tool achieves that, with regard to customer-like machine configurations and Agent setup. Please see [this document](https://datadoghq.atlassian.net/wiki/spaces/SMP/pages/2983952861/Workload+Checks) and its children for more details.

REF SMP-673

blt committed:
[Single Machine Performance] Nightly Workload Checks

This commit introduces the functional test that drives Workload Checks in the
Agent project. Tests for a 'typical' machine and its workloads are also added.

REF SMP-673

Signed-off-by: Brian L. Troutwine <brian.troutwine@datadoghq.com>

scottopell commented on test/workload-checks/typical/cases/apm_app_server/experiment.yaml:
```diff
@@ -0,0 +1,72 @@
+  CPU resource consumption.
+
+teams:
+  - agent-core
```
There was previously an agent team called "agent-core" -- this has now split into "agent-shared-components" and my team "agent-metrics-logs". I assume this is just saying "this is a core agent check", but maybe a different name would be less confusing? Or maybe I'm overthinking, feel free to ship as-is.

blt replied:
Ah yeah, and these aren't hooked up properly yet. I will drop this.

blt replied:
Addressed in [b09e52b](https://github.com/DataDog/datadog-agent/pull/18814/commits/b09e52bbc04d7901437d78dd2e3934031d3f383d)

scottopell commented on test/workload-checks/typical/cases/app_server/experiment.yaml:
```diff
@@ -0,0 +1,70 @@
+      series: "rate(bytes_written['dogstatsd'])"
+      # Lading is configured to send 32 MB/s. This is a relatively high priority
+      # concern by bounding below close to the transmission rate.
+      lower_bound: 0b
```
similar to a previous comment, lower_bound is `0 bytes` which is not "bound close to the transmission rate"

blt replied:
Addressed in [b09e52b](https://github.com/DataDog/datadog-agent/pull/18814/commits/b09e52bbc04d7901437d78dd2e3934031d3f383d)

blt committed:
address PR feedback

Signed-off-by: Brian L. Troutwine <brian.troutwine@datadoghq.com>

blt commented on test/workload-checks/typical/cases/apm_app_server/experiment.yaml:
```diff
@@ -0,0 +1,72 @@
+      # Lading is configured to send 64 MB/s on each of two connections. This is
+      # a relatively high priority concern by bounding below close to the
+      # transmission rate.
+      lower_bound: 2Mb
```
Addressed in [b09e52b](https://github.com/DataDog/datadog-agent/pull/18814/commits/b09e52bbc04d7901437d78dd2e3934031d3f383d)

spencergilbert commented on test/workload-checks/typical/cases/apm_app_server/README.md:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏ Are all three of the READMEs are intended to be identical?

blt replied:
Oops, no, this is a bug.

blt committed:
documentation fixes

Signed-off-by: Brian L. Troutwine <brian.troutwine@datadoghq.com>

blt committed:
Exempt SMP test files from path check

Dinged in the previous commit these paths are test-only and contain deep file
paths that replicate Agent configuration.

Signed-off-by: Brian L. Troutwine <brian.troutwine@datadoghq.com>

blt committed:
python lint ding

Signed-off-by: Brian L. Troutwine <brian.troutwine@datadoghq.com>
